src/metabase/driver/sql_jdbc/sync/interface.clj

```diff
@@ -103,3 +102,25 @@
   {:added "0.43.0", :arglists '([driver database table])}
   driver/dispatch-on-initialized-driver
   :hierarchy #'driver/hierarchy)
+
+(defmulti current-user-table-privileges
+  "Returns the rows of data as arrays needed to populate the table_privileges table
+   with the DB connection's current user privileges.
+   The data contains the privileges that the user has on the given `database`.
+   The privileges include select, insert, update, and delete.
+
+   The rows have the following keys and value types:
+     - role            :- [:maybe :string]
+     - schema          :- [:maybe :string]
+     - table           :- :string
+     - select          :- :boolean
+     - update          :- :boolean
+     - insert          :- :boolean
+     - delete          :- :boolean
+
+   Either:
+   (1) role is null, corresponding to the privileges of the DB connection's current user
+   (2) role is not null, corresponding to the privileges of the role"
+  {:added "0.49.0" :arglists '([driver ^java.sql.Connection connection])}
```

Comment:
Lets leave room for kwargs
```suggestion
  {:added "0.49.0" :arglists '([driver ^java.sql.Connection connection & args])}
```

Reply:
I is intentional not to use `& {:as options}` here, in case we decide to add optional positional arguments too, or is this just a general style preference? I see a few `:arglists` in the codebase that explicitly show it as kwargs and even mention some keys.